ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/rewrite/testtool/templates/Divulgence2.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.rewrite.testtool.templates
+
+import com.daml.ledger.api.rewrite.testtool.infrastructure.LedgerTestContext
+import com.digitalasset.ledger.api.v1.value.Value.Sum.{ContractId, Party}
+
+import scala.concurrent.Future
+
+object Divulgence2 {
+  def apply(party: String, div2Signatory: String, div2Fetcher: String)(
+      implicit context: LedgerTestContext): Future[Divulgence2] =
+    context
+      .create(
+        party,
+        ids.divulgence2,
+        Map("div2Signatory" -> new Party(div2Signatory), "div2Fetcher" -> new Party(div2Fetcher))
+      )
+      .map(new Divulgence2(_, party, div2Signatory, div2Fetcher) {})
+
+}
+
+sealed abstract case class Divulgence2(
+    contractId: String,
+    party: String,
+    div2Signatory: String,
+    div2Fetcher: String) {
```

Comment:
This field doesn't seem to be used anywhere.

Reply:
Yeah, not sure whether this is used in some other test, I can remove it without consequences for now I guess. Ideally I'd like to supplant this with code generation.

Reply:
I guess we'll wait with that until the Divulgence test is complete.

Reply:
Yes, I've just swapped in the new tool, I'll finish `DivulgenceIT` and `ContractKeysIT` (since it's used for conformance tests of the reference DAML-on-X implmentation) and then I'd say we can merge, module further review comments, mostly because this PR is getting huge with the swap and subsequent deduplication. Open points can be addressed one by one in upcoming PRs.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2282/commits/c7e6538d4efaacb3ed283c726acc0a12dfaaa80f